packages/slice-machine/test/src/modules/pushChangesSaga.test.ts

```diff
@@ -1,562 +1,243 @@
+        changesPushSaga,
+        changesPushCreator.request(changesPayload)
+      );
+    });
+    test("Pushes changes when API response is OK display success toaster", () => {
+      saga.next().call(pushChanges, pushChangesApiPayload);
+
+      saga.next(null).call(getState);
+
+      saga.next(dummyServerState).put(
+        refreshStateCreator({
+          env: dummyServerState.env,
+          remoteCustomTypes: dummyServerState.remoteCustomTypes,
+          localCustomTypes: dummyServerState.customTypes,
+          libraries: dummyServerState.libraries,
+          remoteSlices: [],
+          clientError: undefined,
+        })
+      );
+
+      saga.next().put(changesPushCreator.success());
+
+      saga.next().put(
+        openToasterCreator({
+          content: "All slices and custom types have been pushed",
+          type: ToasterType.SUCCESS,
+        })
+      );
+
+      saga.next().isDone();
+    });
+
+    test.each([
+      [LimitType.HARD, ModalKeysEnum.HARD_DELETE_DOCUMENTS_DRAWER],
+      [LimitType.SOFT, ModalKeysEnum.SOFT_DELETE_DOCUMENTS_DRAWER],
+    ])(
+      "Displays delete limit modal when there is a %s limit response",
+      async (limitType, expectedModalKey) => {
+        saga.next().call(pushChanges, pushChangesApiPayload);
+
+        // This test will also verify that the details are sorted in descending order.
+        saga
+          .next({
+            type: limitType,
+            details: {
+              customTypes: [
+                {
+                  id: "CT1",
+                  numberOfDocuments: 2000,
+                  url: "url",
+                },
+                {
+                  id: "CT2",
+                  numberOfDocuments: 3000,
+                  url: "url",
+                },
+              ],
+            },
+          })
+          .put(
+            changesPushCreator.failure({
+              type: limitType,
+              details: {
+                customTypes: [
+                  {
+                    id: "CT2",
+                    numberOfDocuments: 3000,
+                    url: "url",
+                  },
+                  {
+                    id: "CT1",
+                    numberOfDocuments: 2000,
+                    url: "url",
+                  },
+                ],
+              },
+            })
+          );
+
+        saga.next().put(
+          modalOpenCreator({
+            modalKey: expectedModalKey,
+          })
+        );
+
+        saga.next().isDone();
+      }
+    );
+
+    test("Displays an error toaster when there is an API error on push", () => {
+      saga.next().call(pushChanges, pushChangesApiPayload);
+
+      saga.throw(new Error()).put(
+        openToasterCreator({
+          content:
+            "Something went wrong when pushing your changes. Check your terminal logs.",
+          type: ToasterType.ERROR,
+        })
+      );
+
+      saga.next().isDone();
+    });
+
+    // TODO: unskip when auth error is handled
```

Comment:
Do we have a clear list of things left to do somewhere?

Reply:
I'm creating a ticket for this one, and have updated the delete tickets too